src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,342 @@
+/*
+ * This file is part of the Soletta™ Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sol-flow/netctl.h"
+#include "sol-flow-internal.h"
+
+#include <sol-netctl.h>
+#include <sol-str-table.h>
+#include <sol-util-internal.h>
+#include <errno.h>
+
+struct list_service_data {
+    struct sol_ptr_vector service_list;
+    struct sol_flow_node *node;
```

Comment:
no need to keep a node here. Instead, when you need both the private data (list_service_data) and the node, pass the node itself as the `void *data` and then call `sol_flow_node_get_private_data()` (https://github.com/solettaproject/soletta/blob/master/src/lib/flow/include/sol-flow.h#L121)